MuxExoPlayer/src/main/java/com/mux/stats/sdk/muxstats/MuxBaseExoPlayer.java

```diff
@@ -343,6 +346,35 @@ public void onAdsManagerLoaded(AdsManagerLoadedEvent adsManagerLoadedEvent) {
 
   protected abstract boolean isLivePlayback();
 
+  /**
+   * Detects if the runtime ExoPlayer has the HLS extension. If not, references to symbols like
+   * HlsManifest, HlsMediaPlaylist, etc will cause runtime errors
+   */
+  protected final boolean isHlsExtensionAvailable() {
+    if (foundHlsExtension == null) {
+      try {
+        // This class is for sure in the hls extension
+        Class.forName("com.google.android.exoplayer2.source.hls.HlsManifest");
+        foundHlsExtension = true;
+      } catch (ClassNotFoundException e) {
+        MuxLogger.w(TAG, "exoplayer library-hls not available. Some features may not work");
+        foundHlsExtension = false;
+      }
+    }
```

Comment:
Will this stuff work on obfuscated builds?

Reply:
Good point! I updated it to use class references instead. It should crash on `HlsManifest.class` now